src/deluge/gui/views/audio_clip_view.cpp

```diff
@@ -230,57 +276,43 @@ void AudioClipView::graphicsRoutine() {
 	    || playbackHandler.ticksLeftInCountIn) {
 		newTickSquare = 255;
 	}
-
-	// Tempoless or arranger recording
 	else if (!playbackHandler.isEitherClockActive()
 	         || (currentPlaybackMode == &arrangement && getCurrentClip()->getCurrentlyRecordingLinearly())) {
 		newTickSquare = kDisplayWidth - 1;
-
-		// Linearly recording
-		if (getCurrentClip()->getCurrentlyRecordingLinearly()) { // This would have to be true if we got here, I think?
-			getCurrentAudioClip()->renderData.xScroll = -1;      // Make sure values are recalculated
-			needsRenderingDependingOnSubMode();
-		}
 	}
-
 	else {
 		newTickSquare = getTickSquare();
-
-		if (getCurrentAudioClip()->getCurrentlyRecordingLinearly()) {
-			needsRenderingDependingOnSubMode();
-		}
-
 		if (newTickSquare < 0 || newTickSquare >= kDisplayWidth) {
 			newTickSquare = 255;
 		}
 	}
 
 	if (PadLEDs::flashCursor != FLASH_CURSOR_OFF && (newTickSquare != lastTickSquare || mustRedrawTickSquares)) {
-
 		uint8_t tickSquares[kDisplayHeight];
 		memset(tickSquares, newTickSquare, kDisplayHeight);
 
-		const uint8_t* colours = getCurrentClip()->getCurrentlyRecordingLinearly() ? twos : zeroes;
-		PadLEDs::setTickSquares(tickSquares, colours);
+		const uint8_t* coloursPtr = getCurrentClip()->getCurrentlyRecordingLinearly()
+		                                ? (const uint8_t*)"\2\2\2\2\2\2\2\2"
+		                                : (const uint8_t*)"\0\0\0\0\0\0\0\0";
+		PadLEDs::setTickSquares(tickSquares, coloursPtr);
```

Comment:
Please don't do type-casts like this, it's extremely unclear what the data should be and why. If it should be an array, please use an array:

```suggestion
		const std::array<uint8_t, 8> coloursArray = {0};
		if (getCurrentClip()->getCurrentlyRecordingLinearly) {
		    coloursArray.fill(2);
		}
		PadLEDs::setTickSquares(tickSquares, coloursArray.data());
```